flycheck.el

```diff
@@ -3716,8 +3735,14 @@ Return the created overlay."
   ;; We must have a proper error region for the sake of fringe indication,
   ;; error display and error navigation, even if the highlighting is disabled.
   ;; We erase the highlighting later on in this case
-  (pcase-let* ((`(,beg . ,end) (flycheck-error-region-for-mode
-                                err (or flycheck-highlighting-mode 'lines)))
+  (pcase-let* ((`(,beg . ,end)
+                (if (flycheck-relevant-error-other-file-p err)
```

Comment:
(continued) …here it simplifies to having different file names, since the error has already been filtered.  I wondered if it'd be better to inline it in both places and simplify accordingly.

Reply:
No, it doesn't simplify to that - either the error's file or the current buffer's file name could be empty, and duplicating that logic from `flycheck-relevant-error-file-p` therefore didn't feel right here. So I extracted `flycheck-relevant-error-other-file-p`, with the result that that function checks the presence of those two filenames - in the context in which it is called in `flycheck-relevant-error-file-p` that check is redundant.